docs/components/modal.md

```diff
@@ -1,348 +1,752 @@
-      </div>
-    </template>
-    <template v-slot:header-content>
-      <KIcon icon="kong" class="horizontal-spacing" />
-      Welcome!
-    </template>
-    <template v-slot:body-content>Get set up with the 2-step quickstart to see live data pushed through a gateway service within minutes.</template>
-    <template v-slot:action-buttons>
-      <KButton appearance="tertiary" class="horizontal-spacing" @click="slottedIsOpen2 = false">Skip</KButton>
-      <KButton
-        appearance="primary"
-        @click="() => slottedIsOpen2 = false">Onboard me!</KButton>
-    </template>
+    Try pressing Enter.
   </KModal>
-</div>
+</KComponent>
 
 ```html
 <KModal
-  :is-visible="isVisible"
-  title="Welcome!"
-  hide-cancel-button
-  text-align="left"
-  dismiss-button-theme="light"
-  @canceled="isVisible = false"
+  :proceed-on-enter="false"
+  hide-close-icon
+  :visible="modalVisible"
+  @cancel="handleModalClose"
+  @proceed="handleModalProceed"
+>
+  Try pressing Enter.
+</KModal>
+```
+
+### closeOnBackdropClick
+
+Whether clicking on backdrop should close the modal (by emitting the [`cancel` event](#cancel)). Defaults to `false`.
+
+<KComponent
+  v-slot="{ data }"
+  :data="{ closeOnBackdropClick: false }"
 >
-  <template v-slot:header-image>
-    <div class="slot-image-content1">
-      <img src="/img/dark-demo.png" alt="Circuit board" />
+  <div class="vertical-container">
+    <div>
+      <KButton @click="modal13Visible = true">Modal</KButton>
     </div>
-  </template>
-  <template v-slot:header-content>
-    <KIcon icon="kong" />
-    Welcome!
-  </template>
-  <template v-slot:body-content>Get set up with the quickstart to see live data pushed through a gateway service within minutes.</template>
-  <template v-slot:action-buttons>
-    <KButton appearance="tertiary" @click="isVisible = false">Skip</KButton>
-    <KButton
-      appearance="primary"
-      @click="() => isVisible = false">Onboard me!</KButton>
-  </template>
+    <KInputSwitch
+      v-model="data.closeOnBackdropClick"
+      label="Close on backdrop click"
+    />
+  </div>
+
+  <KModal
+    :close-on-backdrop-click="data.closeOnBackdropClick"
+    hide-close-icon
+    :visible="modal13Visible"
+    @cancel="closeAllModals"
+    @proceed="closeAllModals"
+  >
+    Try clicking on modal backdrop.
+  </KModal>
+</KComponent>
+
+```html
+<KModal
+  :close-on-backdrop-click="false"
+  hide-close-icon
+  :visible="modalVisible"
+  @cancel="handleModalClose"
+  @proceed="handleModalProceed"
+>
+  Try clicking on modal backdrop.
+</KModal>
+```
+
+### width
+
+With of the modal window. Defaults to `500px`.
+
+<KButton @click="modal14Visible = true">Modal</KButton>
+<KModal
+  width="90%"
+  title="Modal"
+  :visible="modal14Visible"
+  @cancel="closeAllModals"
+  @proceed="closeAllModals"
+/>
+
+```html
+<KModal
+  width="90%"
+  title="Modal"
+  :visible="modalVisible"
+  @cancel="handleModalClose"
+  @proceed="handleModalProceed"
+/>
+```
+
+### maxHeight
```

Comment:
suggestion: move width and max-height above the action button props since these are likely more widely used